In [1]:
#importing necessary libraries:

import pandas as pd

In [2]:
#To read csv file using pandas:

data = pd.read_csv("Advertising.csv")

In [3]:
#To see first five rows of the dataframe:

data.head()

Unnamed: 0     TV  Radio  Newspaper  Sales
0           1  230.1   37.8       69.2   22.1
1           2   44.5   39.3       45.1   10.4
2           3   17.2   45.9       69.3    9.3
3           4  151.5   41.3       58.5   18.5
4           5  180.8   10.8       58.4   12.9

In [4]:
#To check missing values in the data:

data.isnull().sum()

Unnamed: 0    0
TV            0
Radio         0
Newspaper     0
Sales         0
dtype: int64

In [5]:
#Importing few necessary libraries:

from sklearn.ensemble import RandomForestRegressor
from sklearn.model_selection import train_test_split
from sklearn.metrics import r2_score

In [6]:
#To assign our independent variables and dependent variable into different data frames: 

X = data.drop(["Unnamed: 0", "Sales"], axis=1)
y = data["Sales"]

In [7]:
#To see first five rows of our input variables dataframe:

X.head()

TV  Radio  Newspaper
0  230.1   37.8       69.2
1   44.5   39.3       45.1
2   17.2   45.9       69.3
3  151.5   41.3       58.5
4  180.8   10.8       58.4

In [8]:
#To see first five rows of our output variable dataframe:

y.head()

0    22.1
1    10.4
2     9.3
3    18.5
4    12.9
Name: Sales, dtype: float64

In [9]:
#To split our data into training and testing dataset:

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=1)

In [10]:
#To check the shapes of our variables:

print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)

(160, 3) (40, 3) (160,) (40,)


In [11]:
#Building classification model using Random Forest:

model = RandomForestRegressor(max_depth=16, criterion='absolute_error', random_state=15)

In [12]:
#Fitting the model:

model.fit(X_train, y_train)

RandomForestRegressor(criterion='absolute_error', max_depth=16, random_state=15)

In [13]:
#Predicting the output variable for testing dataset:

y_pred = model.predict(X_test)

In [14]:
#To check the good fit of our model using r2:

print(r2_score(y_pred, y_test))

0.9801535308203952


#### Here we are getting r2 as 0.98 on test data, which is considered good. Hence we can say that our model is a good fit for given dataset.